MaxText/train.py

```diff
@@ -314,22 +338,22 @@ def train_loop(config, state=None):
     raise ValueError("Profiling requested but initial profiling step set past training final step")
   last_profiling_step = np.clip(first_profiling_step + config.profiler_steps - 1, first_profiling_step, config.steps - 1)
 
-  nextrng = jax.random.fold_in(init_rng, start_step)
-  example_batch = load_next_batch(data_iterator, None, config)
```

Comment:
A tiny question, does this line change the dataloading? Previously, we load first batch before entering the loop and now inside the loop. I remember there was a PR relevant.